asyncstdlib/itertools.py

```diff
@@ -141,35 +141,38 @@ async def accumulate(iterable, function, *, initial):
             yield value
 
 
-async def chain(*iterables: AnyIterable[T]) -> AsyncIterator[T]:
+class chain(Generic[T], AsyncIterator[T]):
```

Comment:
Since the `AsyncIterator` is already `Generic`, you don't need the latter.
```suggestion
class chain(AsyncIterator[T]):
```